app/CocoaPods/CPMenuRepoUpdateDirector.swift

```diff
@@ -0,0 +1,215 @@
+import Cocoa
+
+class CPMenuRepoUpdateDirector: NSObject {
+  let repoCoordinator = CPSourceRepoCoordinator()
+
+  // The thing that you click on, in the main menu
+  @IBOutlet weak var sourceReposMenuItem: NSMenuItem!
+
+  // The representation of the things that would show when clicked
+  @IBOutlet weak var sourceReposMenu: NSMenu!
+
+  override func awakeFromNib() {
+    super.awakeFromNib()
+
+    // Don't let it be clickable when there's nothing to show
+    sourceReposMenuItem.enabled = false
+    sourceReposMenu.removeAllItems()
+
+    // Once everything's finished then we start looking for repo sources,
+    // and re-run it after every pod install/update, which may have
+    // added new sources
+
+    for key in [NSApplicationDidFinishLaunchingNotification, "CPInstallCompleted"] {
```

Comment:
We might want to make this notification key a static string so that we don't have to worry about typos when using it.